src/go/rpk/pkg/cli/topic/mount.go

```diff
@@ -0,0 +1,108 @@
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var alias string
+	var aliasNamespace string
+	var namespace string
+	var location string
+	var timeout int
+
+	cmd := &cobra.Command{
+		Use:   "mount [TOPICS...]",
+		Short: "Mount a topic to the Redpanda Cluster from tiered storage",
+		Long:  helpMount,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			pf, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			config.CheckExitCloudAdmin(pf)
+			adm, err := adminapi.NewClient(fs, pf)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			topic := rpadmin.InboundTopic{
+				SourceTopic: rpadmin.Topic{
+					Topic:     topics[0],
+					Namespace: namespace,
+				},
+			}
+			if alias != "" && aliasNamespace != "" {
+				topic.Alias = &rpadmin.Topic{
+					Topic:     alias,
+					Namespace: aliasNamespace,
+				}
+			}
+			if location != "" {
+				topic.Location = location
+			}
+
+			migration, err := adm.AddInboundMigration(cmd.Context(), rpadmin.InboundMigration{
+				MigrationType:  "inbound",
+				Topics:         []rpadmin.InboundTopic{topic},
+				ConsumerGroups: []string{}, // not implemented yet
+			})
+			out.MaybeDie(err, "unable to add inbound migration: %v", err)
+
+			tmt := time.Duration(timeout * int(time.Minute/time.Nanosecond))
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), migration.ID, adm, rpadmin.PrepareMigrationAction, "prepared", tmt); errP != nil {
+				out.MaybeDie(errP, "unable to prepare migration", errP)
+			}
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), migration.ID, adm, rpadmin.ExecuteMigrationAction, "executed", tmt); errP != nil {
+				out.MaybeDie(errP, "unable to execute migration", errP)
+			}
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), migration.ID, adm, rpadmin.FinishMigrationAction, "finished", tmt); errP != nil {
+				out.MaybeDie(errP, "unable to finish migration", errP)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&namespace, "namespace", "n", "", "Namespace for the mounted topic (optional)")
+	cmd.Flags().StringVarP(&location, "location", "l", "", "Location (optional)") // need more info on this
+	cmd.Flags().StringVarP(&alias, "alias", "a", "", "New name for the mounted topic, if different from the original (optional)")
+	cmd.Flags().StringVarP(&aliasNamespace, "aliasnamespace", "s", "", "Namespace for the alias topic, if using an alias (optional)")
+	cmd.Flags().IntVarP(&timeout, "timeout", "", 5, "Timeout in minutes for the migration to finish (default: 5)")
+	return cmd
+}
+
+var helpMount = `Mount a topic from tiered storage to the Redpanda cluster.
+
+This command mounts a topic in the Redpanda Cluster using log segments stored in tiered storage.
```

Comment:
@Deflaimun Do we have an article/blog that we can use for our long help text here?

Reply:
That would honestly be great. I put something together as a placeholder/in lieu of a better alternative.

Reply:
Let me check with the team

Reply:
We have the general page for Tiered Storage https://docs.redpanda.com/current/manage/tiered-storage/ that is packed with info. 

This may be overkill. Are you guys looking for something more specific? 
(Still asking the team if we have something specific in the works about this new feature itself)